packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -337,6 +337,10 @@ mixin SchedulerBinding on BindingBase {
   ///
   /// The preferred way to watch for changes to this value is using
   /// [WidgetsBindingObserver.didChangeAppLifecycleState].
+  ///
+  /// Be aware that the [lifecycleState]'s value when your entrypoint is
+  /// executed is undefined since a Flutter instance can be launched at any
+  /// time.
```

Comment:
To implement these parts of the style guide:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-the-passive-voice-recommend-do-not-require
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-empty-prose
...how about:
```suggestion
  ///
  /// The [lifecycleState]'s value when the Dart entrypoint (typicially `void main()`) is
  /// executed is undefined, because a Flutter instance can be launched at any
  /// time.
```

But is "undefined" actually true? Isn't it defined as "whatever the state of the app is at time X"? (For some definition of X that may not be "now" but may be, say, when the instance was launched a few milliseconds ago?)

Reply:
"Undefined" might be a loaded term.  It's not that it isn't assigned a value, but it doesn't have a defined value.  What value it is is non-deterministic.  Do you prefer "not guaranteed", or "non-deterministic"?

The crux is we don't want them to put assumptions in their code about its value in the entry point, but reading what value it is may be useful.

Reply:
I opted for "not guaranteed", let me know how that looks.